pipeline-scripts/release.groovy

```diff
@@ -442,6 +442,37 @@ for f in *.tar.gz *.bz *.zip *.tgz ; do
 done
     ''')
 
+    if ( minor > 0 && (client_type == "ocp-dev-preview" || release_name.indexOf('-fc.') > -1 || release_name.indexOf('-rc.') > -1 ) {
+        // To encourage customers to explore dev-previews & pre-GA releases, populate changelog
+        // https://issues.redhat.com/browse/ART-3040
+        prevMinor = minor - 1
+        rcURL = commonlib.getReleaseControllerURL(release_name)
+        rcArch = commonlib.getReleaseControllerArch(release_name)
+        stableStream = (rcArch=="amd64")?"4-stable":"4-stable-${rcArch}"
+        prevGA = "${major}.${prevMinor}.0"
+        outputDest = "${BASE_TO_MIRROR_DIR}/changelog.html"
+
+        // See if the previous minor has GA'd yet; e.g. https://amd64.ocp.releases.ci.openshift.org/releasestream/4-stable/release/4.8.0
+        def check = httpRequest(
+            url: "${rcURL}/releasestream/${stableStream}/release/${prevGA}",
+            httpMode: 'GET',
+            validResponseCodes: '200,404',  // if we get 404, do not compute changelog yet; prev has not GA'd.
```

Comment:
won't this cause an exception for any other problems? could we try/catch/retry in case of network bobbles etc? in general it seems like it would be bad for computing the changelog to nuke the release...

Reply:
I did want it to get some attention if it started to fail, but nuking the release is too much. I've added a try/catch with a slack on failure.